fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3303,8 +3325,17 @@ void DDTeamCollection::updateAvailableSpacePivots() {
 		teamPivots.pivotAvailableSpaceRatio =
 		    std::max(SERVER_KNOBS->MIN_AVAILABLE_SPACE_RATIO,
 		             std::min(SERVER_KNOBS->TARGET_AVAILABLE_SPACE_RATIO, teamAvailableSpace[pivot]));
+
+		size_t strictPivot =
+		    teamAvailableSpace.size() * std::min(1.0, SERVER_KNOBS->DD_STRICT_AVAILABLE_SPACE_PIVOT_RATIO);
+		std::nth_element(teamAvailableSpace.begin(),
+		                 teamAvailableSpace.begin() + strictPivot,
+		                 teamAvailableSpace.end(),
+		                 std::greater{});
+		teamPivots.strictPivotAS = teamAvailableSpace[strictPivot];
```

Comment:
We should avoid doing this extra computation (also for `strictPivotCPU` below) if `SERVER_KNOBS->DD_REEVALUATION_ENABLED` is false. (Ideally, turning off a knob should turn off as much computation and behavior change for the feature as is reasonable.)

Reply:
Okay, i've guarded these computation by `DD_REEVALUATION_ENABLED`